packages/cluster-browser/src/Cluster/joinCluster.ts

```diff
@@ -0,0 +1,163 @@
+import { map, mergeAll, tap } from 'rxjs/operators';
+import { from, Observable, pipe, Subject } from 'rxjs';
+import { ClusterEvent, ClusterOptions, JoinCluster, MembersData } from '@scalecube/api/lib/cluster';
+import { getFullAddress } from '@scalecube/utils';
+import { createServer, send as sendAddress, on } from './connection';
+import { Address } from '@scalecube/api';
+
+function debug(ctx: Context, logname: string, filter: any, ...log: any[]) {
+  if (filter(ctx, logname, log)) {
+    // tslint:disable-next-line:no-console
+    console.log(ctx.address, logname, ...log);
+  }
+}
+// const DEBUG = (ctx: Context, name: string,...log: any[]) => debug(ctx, name, ()=>{
+//     return ctx.address === 'pm://defaultHost:8080/b' &&
+//         name === 'notify';
+// }, ...log);
+const DEBUG = (...arg: any[]) => {};
+
+function sendInit(ctx: Context, seed: string) {
+  ctx.send(seed, {
+    type: 'INIT',
+    from: ctx.address,
+    sender: ctx.address,
+    items: ctx.membersData[ctx.address],
+  } as ClusterEvent & { sender: any });
+}
+
+function notifyMembersChanged(ctx: Context, send: any, to: string) {
+  ctx.members$.subscribe((member) => {
+    DEBUG(ctx, 'notify', to, member);
+    if (member.from !== to && member.sender !== to) {
+      send({
+        ...member,
+        type: member.type === 'REMOVED' ? 'REMOVED' : 'ADDED',
+        sender: ctx.address,
+      });
+    }
+  });
+}
+
+function sendMembers(ctx: Context, send: any, to: string) {
+  DEBUG(ctx, 'send', to, ctx.membersData);
+  Object.keys(ctx.membersData).forEach((address) => {
+    if (address !== to) {
+      send({
+        type: 'ADDED',
+        from: address,
+        items: ctx.membersData[address],
+        sender: ctx.address,
+      });
+    }
+  });
+}
+
+function updateMembers(ctx: Context, e: InternalClusterEvent) {
+  DEBUG(ctx, 'update', e);
+  if (!!ctx.membersData[e.from] === (e.type === 'REMOVED')) {
+    // DEBUG(options.address.path, msg.data, membersData[msg.data.from]);
+    ctx.members$.next(e);
+  }
+  switch (e.type) {
+    case 'INIT':
+      if (e.from !== e.sender) {
```

Comment:
why you need this condition?
```suggestion
      if (e.from !== e.sender) {
```

Reply:
LGTM

Reply:
Is it clear why? "why you need this condition?"
@katzIdo